evalml/model_understanding/graphs.py

```diff
@@ -555,3 +555,31 @@ def graph_prediction_vs_actual(y_true, y_pred, outlier_threshold=None):
                                 marker=_go.scatter.Marker(color=color),
                                 name=name))
     return _go.Figure(layout=layout, data=data)
+
+
+def graph_prediction_vs_target_over_time(pipeline, X, y, dates):
```

Comment:
I commented about this on the issue, but to reiterate: its fine to leave this as-is and expect `dates` as a 4th arg, but heads up that @angela97lin is in the process of standardizing all the util/graph methods to use `DataTable`s, in which case `X.time_index` should get you the dates.